### ISSUE TITLE
Misc: Explore some way to import data from other sources

### ISSUE THREAD

**Issue by mittal-parth** [mittal-parth/personal-portfolio#60]
Currently, one needs to edit the [constants/index.js](https://github.com/mittal-parth/personal-portfolio/blob/main/src/constants/index.js) file for all the info. This could be quite a task as it involves a lot of data.

The quickest way that I have seen is to copy your info from an existing site (like LinkedIn) and ask an LLM to convert it in our required JSON format. 

However, if we can explore some solution like an 'Import from LinkedIn' solution, it would be nice. 
More providers like Devfolio can be added. 

This issue is open for discussion and does not necessarily have a defined outcome.

**Comment by A-4-Atom** [comment]
Many websites provide a way to download your data in form of _JSON_ or _CSV_. I was thinking we could have the user place that file in root folder and run a command like **npm run generate-data**, which would run a script that uses a LLM to convert data in our format.

But I am a little confused how would we use the LLM via api? It would require the api key to do any task. Do we ask the user to add an api key as well?

What do you think?
@mittal-parth

**Comment by mittal-parth** [comment]
Hey @A-4-Atom!

I think the issue is not the LLM but the export option itself. I feel the main source of info for most ppl is LinkedIn. Does LinkedIn have such a feature? (I am not aware of one)

If it does allow export in any format, converting it to our JSON file doesn't require us to add an LLM call. The user can copy that and ask any LLM to convert it to our format.

**Comment by A-4-Atom** [comment]
@mittal-parth Yes, LinkedIn allows users to download the data, though it's not instant and can take up to 24 hours. I'm gonna try downloading my data and see if chatgpt can convert it successfully to desired format.

**Comment by A-4-Atom** [comment]
@mittal-parth
 Hi, I downloaded my data from linkedIn and asked LLM's to convert it to our format. ChatGPT gave inconsistent results, but deepseek was better. It was able to read csv files and modify data accordingly, also it skipped the parts where there's no data in csv. 

Users will also need to update the imports and images themselves in many places like projects or achievements. Do you think it's worth the hassle? would it be easier to just type the data as linkedin takes 1-2 days to give full data.

**Comment by mittal-parth** [comment]
Hey @A-4-Atom 

Interestingly, even I got my archive today :)

It does actually have everything. I was previously just exporting the profile info which is just one line xD

Building a consistent, programmatic CSV parser is easy actually. No need of an LLM also in this case. One can specify their directory for their files and we do the rest since the LinkedIn output will always be in the same format. 

Coming to the picture files, it does have one CSV containing the 'Rich_Media'. This has all the files uploaded. However, there are two issues here:

1. Unless you add a media description (which I assume ppl dont, since I myself havent added to any surprisingly), there is just the link, nothing else. What is useful where, is difficult in this case. 
2. It wont have company/org/education logos. This is actually the most important and that isnt there. To overcome this, we could utilise some API that returns some image based on certain parameters like name, without bg,etc. But again, this will be slightly inconsistent/in-exhaustive. 

I don't how useful this would be honestly, but does reduce the friction.

**Comment by A-4-Atom** [comment]
@mittal-parth 
What if we parse the data from csv and fill as much fields as possible.
And use placeholder image/icons so that the user knows exactly what image it needs by looking at the portfolio.
Even if the user have to manually add images, they save time on typing and organizing content.

**Comment by A-4-Atom** [comment]
@mittal-parth I'll check it out. I would've done it myself but college is taking too much of my time right now.
